src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -477,9 +505,47 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
         {
             // Called outside of async state machine to propagate certain exception even without awaiting the returned task.
             CheckRequestBeforeSend(request);
-
             (CancellationTokenSource cts, bool disposeCts, CancellationTokenSource pendingRequestsCts) = PrepareCancellationTokenSource(cancellationToken);
-            return SendAsyncCore(request, completionOption, async: true, cts, disposeCts, pendingRequestsCts, cancellationToken).AsTask();
+
+            return Core(request, completionOption, cts, disposeCts, pendingRequestsCts, cancellationToken);
+
+            async Task<HttpResponseMessage> Core(
```

Comment:
static?

Reply:
It uses members of `this`.  Unforunately there's no more fine-grained capture syntax to let you say "let me capture `this` but nothing else".